platform/helpers.go

```diff
@@ -16,13 +17,21 @@ import (
 
 // Private Helpers
 
-func getCurrentUsername() (string, error) {
+func getCurrentUsername() (username string, err error) {
+
 	user, err := user.Current()
 	if err != nil {
 		return "", err
 	}
 
-	username := user.Username
+	reg, err := regexp.Compile("[^a-zA-Z0-9]+")
+	if err != nil {
+		return "", err
+	}
+
+	username = reg.ReplaceAllString(user.Username, "")
+	username = username[:12]
```

Comment:
`brave init` fails here if username is <12 characters long.

Suggesting taking argmin (len(username), 12)

Reply:
Thanks - too much Python. Pushed change.